.github/workflows/actions/deploy-environment/action.yml

```diff
@@ -0,0 +1,74 @@
+name: Deploy PAAS environment
+
+inputs:
+  environment_name:
+    description: 'The name of the environment'
+    required: true
+  docker_image:
+    description: 'The Docker image to deploy to the environment'
+    required: true
+  azure_credentials:
+    description: 'JSON object containing a service principal that can read from Azure Key Vault'
+    required: true
+  terraform_vars:
+    description: 'Path to the tfvars file for the environment'
+    required: true
+  terraform_backend_vars:
+    description: 'Path to the backend tfvars file for the environment'
+    required: true
+
+outputs:
+  environment_url:
+    description: 'The base URL for the deployed environment'
+    value: ${{ steps.terraform.outputs.url }}
+
+runs:
+  using: composite
+
+  steps:
+    - name: Get Key Vault name
+      id: keyvault
+      run: |
+        KEY_VAULT_NAME=$(cat $TFVARS | grep -E '^key_vault_name' | cut -d'=' -f2 | tr -d '"' | tr -d ' ')
```

Comment:
tfvars.json may also be used to store terraform variables so they can be read with `jq`
example: https://github.com/DFE-Digital/register-trainee-teachers/blob/main/.github/workflows/deploy.yml#L83

Reply:
Converted the file to JSON and updated this script to use `jq`.